wurst/math/Matrices.wurst

```diff
@@ -0,0 +1,418 @@
+    // Scaling matrix
+    let v = vec2(0.5, 0.25)
+    print("{0} to scaling {1}\n".format(v.toString(), v.toScaling().toString()))
+    // Rotation
+    let a = angle(45*DEGTORAD)
+    print("Rotation by {0} degrees is {1}\n".format(a.degrees().toString(), a.toMatrix().toString()))
+
+
+function test33()
+    print("=============== Matrix 3x3 testing ===============\n")
+    var matrix = mat3(  1, 2, 3,
+                        4, 5, 6,
+                        7, 8, 9)
+    print("Base {0}\n".format(matrix.toString()))
+    // Determinant
+    print("Determinant is {0}\n".format(matrix.determinant().toString()))
+    // Inverse
+    var inverse = matrix.inverse()
+    print("Inverse matrix exists: {0}\n".format((inverse.success.toString())))
+    // Doesnt exist then new one
+    var tmp = mat3( 10, 2, 3,
+                    4, 5, 6,
+                    7, 8, 9)
+    print("Try another {0}\n".format(tmp.toString()))
+    inverse = tmp.inverse()
+    print("Inverse matrix exists: {0}\n".format((inverse.success.toString())))
+    // Inverse once again
+    print("Inverse {0}\n".format(inverse.matrix.toString()))
+    // Inverse
+    var inverseBack = inverse.matrix.inverse()
+    print("Inverse from inverse matrix (base expected): {0}\n".format(inverseBack.matrix.toString()))
+    // Test inverse matrix
+    var mult = tmp*inverse.matrix
+    print("Base * Inverse (indetity expected) Result: {0}\n".format((mult.toString())))
+    // Return to base matrix
+    print("Return to base {0}\n".format(matrix.toString()))
+    // Get columns
+    print("Columns by id (-1, 0, 1, 2, 3):")
+    for i=-1 to 3
+        print(matrix.col(i).toString())
+    // Get rows
+    print("\nRows by id (-1, 0, 1, 2, 3):")
+    for i=-1 to 3
+        print(matrix.row(i).toString())
+    // Transpose
+    print("\nTranspose {0}\n".format(matrix.transpose().toString()))
+    // + itself
+    print("Matrix + Matrix = {0}\n".format((matrix + matrix).toString()))
+    // - itself
+    print("Matrix - Matrix = {0}\n".format((matrix - matrix).toString()))
+    // * itself
+    print("Matrix * Matrix = {0}\n".format((matrix*matrix).toString()))
+    // * scalar
+    let scalar = 15.
+    print("Matrix * {0} = {1}\n".format(scalar.toString(), (matrix*scalar).toString()))
+    // Scaling matrix
+    let v = vec3(0.5, 0.25, 0.75)
+    print("{0} to scaling {1}\n".format(v.toString(), v.toScaling().toString()))
+    // Rotation
+    let axis = vec3(1, 0, 0)
+    let a = angle(90*DEGTORAD)
+    let rotation = axis.toMatrix(a)
+    print("Rotation around {0} by {1} degrees is rotation {2}\n".format(axis.toString(), a.degrees().toString(), rotation.toString()))
+    // To Euler angles
+    // Cannot produce this tests bcz Asin is not implemented yet.
+    let euler = rotation.toEuler()
+    print("Found rotation matrix to Euler angles {0}".format(euler.toString()))
+    
+
+@test
+function test()

```

Comment:
What else should Iknow to not rewrite it again again and again. How many instructions do you have and why I can't see the whole list?

Reply:
You could check out existing packages before doing something very different.

There is no list _yet_, but I also thought about it being useful in the future. Should probably start a code conventions doc.

Reply:
And I did. For example, `Vectros` package contains `*` with space around it and without or `Angles` package. But  have to.

Reply:
I'm talking about the tests. Obviously many packages are old, made or modified by different people and, resulting in different styles. Especially before we had proper PRs and reviews.
New additions however, should follow the convention.
Soon the formatting should be standardized, if @karlek finishes https://github.com/wurstscript/WurstScript/pull/620 and then we can easily format the stdlib consistently.

Reply:
> New additions however, should follow the convention. 

Which doesn't exist.
Dunno how many surprises are waiting for me. Maybe later.

Reply:
I'm sorry to hear you are disappointed by the pull request process, but I don't think it's anything out of the ordinary.
It can easily take weeks before a pull request is accepted, on any project.Just check other PRs on our projects and other repositories.

I really don't understand why you get so frustrated. We appreciate your contribution, just take a little time to do improvements. This isn't critical.

Reply:
May be because I hardly distinguish speech styles but not, I'm not frustrated.
I'm not interested in WC3 much enough to keep going hint2hint in a black room so I decided to  close this until there's the convention and correct list of requirements. As I said, maybe latter.